### PR TITLE
Available CRDs check feature

### PR DESCRIPTION
**Pull request by rewantsoni**
## Changes

- Update the storageClaim controller to react whenever a create/delete event has been enqueued for a specific CRD that's become newly available.
- Exit the ocs-client-operator process with a specific exit code whenever a new crd of interest is being created or deleted
- Catch the exit code with a bash script and restart the ocs-client-operator process
- Conditionally watch the DRClusterConfig CRD 

In the future we will need to watch Maintenance Mode CRD by ramen as well

This PR just adds the base for watching the DRClusterConfig CRD by ramen, the actual creation will be added in https://github.com/red-hat-storage/ocs-client-operator/pull/177

Ref: https://github.com/red-hat-storage/ocs-operator/pull/2745

### CODE REVIEW

**Review by leelavg — COMMENTED** [review]
minor comment rest is same as the one implemented in ocs-operator, isn't it? btw, are you planning on sending another PR for using this mechanism for storagecluster cr?

asking because, I'm going to remove specific code in a couple of days (when we get RC).

File: internal/controller/storageclaim_controller.go [hunk 1]
```diff
@@ -56,6 +59,8 @@ const (
 
 	pvClusterIDIndexName  = "index:persistentVolumeClusterID"
 	vscClusterIDIndexName = "index:volumeSnapshotContentCSIDriver"
+
+	DRClusterConfigCRDName = "drclusterconfigs.ramendr.openshift.io"
```
[the remark this answers is not in the document]
Comment (rewantsoni) [reply internal/controller/storageclaim_controller.go]: No this doesn't need to be exported updated it

**Comment by rewantsoni** [comment]
>minor comment rest is same as the one implemented in ocs-operator, isn't it? btw, are you planning on sending another PR for using this mechanism for storagecluster cr?
asking because, I'm going to remove specific code in a couple of days (when we get RC).


No, I don't plan to send a PR to use this for storageCluster CR

**Review by leelavg — APPROVED** [review]

**Pull request merged**